oem/oem.go

```diff
@@ -0,0 +1,62 @@
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+package oem
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/stmcginnis/gofish"
+	"github.com/stmcginnis/gofish/oem/dell"
+	"github.com/stmcginnis/gofish/redfish"
+)
+
+const (
+	DellRedfishOem = "Dell"
+	Dell           = "Dell"
+	HpeRedfishOem  = "Hpe"
+	Hpe            = "Hpe"
+	ZTRedfishOem   = "Ami"
+	ZT             = "zt"
+)
+
+type EventService struct {
+	redfish.EventService
+}
+
+// GetRedfishVendor queries the redfish root for the OEM field and returns the vendor.
+func GetRedfishVendor(es *redfish.EventService) (string, error) {
+	serviceRoot, err := gofish.ServiceRoot(es.Client)
+	if err != nil {
+		return "", fmt.Errorf("failed to GET redfish service root due to: %w", err)
+	}
+
+	oem := string(serviceRoot.Oem)
+
+	switch {
+	case strings.Contains(oem, DellRedfishOem):
+		return Dell, nil
+	case strings.Contains(oem, HpeRedfishOem):
+		return Hpe, nil
+	case strings.Contains(oem, ZTRedfishOem):
+		return ZT, nil
+	default:
+		return "", fmt.Errorf("failed to match vendor from output: %v", oem)
```

Comment:
It's possible for someone to "bring their own" OEM library (which was actually my original intention).

This is a decent idea to have a factory for it, but the user would still need to look up which vendor is being used to be able to tell what library and objects they have. The return on line 51 below is a problem. It works now because you are only returning Dell, but there doesn't appear to be a good way to have this return whatever vendor specific OEM implementation it is for any of the included OEMs.

It's an interesting idea, but I'd rather keep the library simple and put the OEM special handling on the consumer since there are a lot of decisions and other concerns they will need to make and be aware of when accessing the OEM-specific things.